persistence/rest/src/main/java/org/infinispan/persistence/rest/RestStore.java

```diff
@@ -128,6 +125,7 @@ protected void initChannel(SocketChannel ch) {
       } catch (EncoderException e) {
       }
       this.metadataHelper = Util.getInstance(configuration.metadataHelper(), ctx.getCache().getAdvancedCache().getClassLoader());
+      new HttpResponseHandler(); // Initialize the Netty TypeParameterMatcher
```

Comment:
I have put a log in the description of https://issues.jboss.org/browse/ISPN-7602
I know this looks disgusting, but the REST store really needs an overhaul to use pooling (ISPN-7601) which would trigger proper initialization in the right place.